sdk/ai/ai-inference-rest/test/public/tracing.spec.ts

```diff
@@ -0,0 +1,626 @@
+    let toolCallAnswer = "";
+    let awaitingToolCallAnswer = true;
+    const messages: ChatRequestMessage[] = [
+      {
+        role: "user",
+        content: "What's the weather like in Boston?",
+      },
+      {
+        content: null,
+        role: "assistant",
+        tool_calls: [
+          {
+            function: {
+              arguments: '{"location":"Boston, MA"}',
+              name: "get_current_weather",
+            },
+            id: "call_YZo4xi315MOSB4pt450S2nyR",
+            type: "function",
+          },
+        ],
+      },
+      {
+        role: "tool",
+        content: "The temperature in Boston, MA is 72 degrees fahrenheit",
+        tool_call_id: "call_YZo4xi315MOSB4pt450S2nyR",
+      },
+    ];
+
+    let response: GetChatCompletions200Response | GetChatCompletionsDefaultResponse | undefined;
+    while (awaitingToolCallAnswer) {
+      response = await client.path("/chat/completions").post({
+        body: {
+          messages,
+          tools: [
+            {
+              type: "function",
+              function: getCurrentWeather,
+            },
+          ],
+        },
+        tracingOptions: { tracingContext: context.active() },
+      });
+
+      if (isUnexpected(response)) {
+        throw response.body.error;
+      }
+
+      const stream = response.body;
+      if (!stream) {
+        throw new Error("The response stream is undefined");
+      }
+
+      if (response.status !== "200") {
+        throw new Error("Failed to get chat completions.");
+      }
+
+      const functionArray: Array<any> = [];
+
+      for (const choice of response.body.choices) {
+        const toolCallArray = choice.message?.tool_calls;
+
+        if (toolCallArray) {
+          if (toolCallArray[0].function?.name) {
+            // Include original response from assistant requesting tool call in chat history
+            choice.message.role = "assistant";
+            messages.push(choice.message);
+          }
+          updateToolCalls(toolCallArray, functionArray);
+        }
+        if (choice.finish_reason == "tool_calls") {
+          const messageArray = handleToolCalls(functionArray);
+          messages.push(...messageArray);
+        } else {
+          if (choice.message?.content && choice.message.content != "") {
+            toolCallAnswer += choice.message?.content;
+            awaitingToolCallAnswer = false;
+          }
+        }
+      }
+    }
+    return { messages, response };
+  }
+
+  beforeEach(async (context) => {
+    recorder = await createRecorder(context);
+    client = await createModelClient("completions", recorder);
+    instrumenter = new MockInstrumenter();
+    useInstrumenter(instrumenter);
+  });
+
+  afterEach(async () => {
+    await recorder.stop();
+  });
+
+  it("client test", function () {
+    assert.isNotNull(client);
+    assert.isNotNull(client.path);
+    assert.isNotNull(client.pipeline);
+  });
+
+  it.only("tracing should work", async function () {
```

Comment:
Only kept the basic test here, once you "productionalize" this code you can fix the tests to your liking